Index: sphinx/thirdparty/sphinxapi.php
===================================================================
--- sphinx/thirdparty/sphinxapi.php	(revision 91804)
+++ sphinx/thirdparty/sphinxapi.php	(working copy)
@@ -497,7 +497,7 @@
 
 	function _Send ( $handle, $data, $length )
 	{
-		if ( feof($handle) || fwrite ( $handle, $data, $length ) !== $length )
+		if ( /* feof($handle) || */ fwrite ( $handle, $data, $length ) !== $length )
 		{
 			$this->_error = 'connection unexpectedly closed (timed out?)';
 			$this->_connerror = true;
